tests/integration/helpers/helpers.py

```diff
@@ -281,3 +281,26 @@ async def deploy_and_relate_application_with_pgbouncer_bundle(
     )
 
     return relation
+
+
+async def scale_application(ops_test: OpsTest, application_name: str, count: int) -> None:
+    """Scale a given application to a specific unit count.
+
+    Args:
+        ops_test: The ops test framework instance
+        application_name: The name of the application
+        count: The desired number of units to scale to
+    """
+    async with ops_test.fast_forward():
+        change = count - len(ops_test.model.applications[application_name].units)
+        if change > 0:
+            await ops_test.model.applications[application_name].add_units(change)
+        elif change < 0:
+            units = [
+                unit.name
+                for unit in ops_test.model.applications[application_name].units[0:-change]
+            ]
+            await ops_test.model.applications[application_name].destroy_units(*units)
+        await ops_test.model.wait_for_idle(
```

Comment:
if count == 0, wait_for_idle will never resolve as i believe it'll never meet the condition of being idle